desktop/cmp/button/AppMenuButton.js

```diff
@@ -19,15 +19,32 @@ export const [AppMenuButton, appMenuButton] = hoistCmp.withFactory({
     render(props) {
         let {className, hideOptionsItem, hideFeedbackItem, hideThemeItem, hideAdminItem, hideImpersonateItem, hideLogoutItem, extraItems, ...rest} = props;
         extraItems = extraItems ?
-            [...extraItems.map(m => menuItem(m)), menuDivider()] :
+            [
+                ...extraItems.map(m => {
+                    if (m == '-') return menuDivider();
+                    return React.isValidElement(m) ? m : menuItem(m);
+                }),
+                menuDivider()
+            ] :
             [];
 
+
         hideAdminItem = hideAdminItem || !XH.getUser().isHoistAdmin;
         hideImpersonateItem = hideImpersonateItem || !XH.identityService.canImpersonate;
         hideLogoutItem = hideLogoutItem || XH.appSpec.isSSO;
         hideOptionsItem = hideOptionsItem || !XH.acm.optionsDialogModel.hasOptions;
+        extraItems = extraItems.filter((it, idx, arr) => {
+            if (it.type.name === 'MenuDivider') {
+                // Remove starting separators
+                if (idx === 0) return false;
+
+                // Remove consecutive separators
+                const prev = idx > 0 ? arr[idx - 1] : null;
+                if (prev?.type.name === 'MenuDivider') return false;
+            }
+            return true;
+        });
```

Comment:
This will take care of manually added consecutive dividers, but the way the code is for the hard coded items below, it's possible that all of the menuItems in a given section will be 'omit'ted and will then result in double dividers. Do we want to move away from using 'omit' and manually build the items to avoid this?

Reply:
Why not concatenate extra items and the default items below first, and then run the code to remove consecutive separators on that?

Reply:
i.e. exactly what you suggested :)

Reply:
Yeah will also require the extra step of not including 'omitted' items, since we wont get that magic here